Fix slider thumb not moving

grokys opened the pull request:
## What does the pull request do?

Turns out that #11015 was a problem with layout introduced in #10864: as well as skipping layout for invisible controls, that PR also introduced a change to skip arrange for controls with an invalid measure. This change is a good change, but it introduced a problem in the following case:

- Child invalidates parent measure in arrange pass
- Parent is added to measure & arrange queues
- Arrange pass dequeues parent
- Measure is not valid so parent is not arranged
- Parent is measured
- Parent has been dequeued from arrange queue so no arrange is performed

This PR fixes this by adding controls to the arrange queue after they've been measured, not on measure invalidation.

Also adds a unit test for this case and extra integration tests for `Slider` to catch problems like this.

## Fixed issues

Fixes #11015

grokys committed:
Added ThumbAutomationPeer.

To make slider thumb visible to automation.

grokys committed:
Improve slider integration tests.

They are now failing due to #11015.

grokys committed:
Added failing test for #11015.

Problem is in layout manager.

grokys committed:
Enqueue for arrange after measure.

Fixes the problem described in `Child_Can_Invalidate_Parent_Measure_During_Arrange`. To do this, add controls to the arrange queue after they've been measured, not on measure invalidation.

Fixes #11015

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0033403-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

maxkatz6 committed:
Merge branch 'master' into fixes/11015-slider-thumb-not-moving

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0033425-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]

grokys committed:
Fix tests on macOS.

`MoveByOffset` caused Appium to hang, but `MoveToElement` with an offset seems to work fine.

avaloniaui-team commented:
You can test this PR using the following package version. `11.0.999-cibuild0033466-beta`. (feed url: https://pkgs.dev.azure.com/AvaloniaUI/AvaloniaUI/_packaging/avalonia-all/nuget/v3/index.json) [PRBUILDID]
